cs-excel/src/main/java/io/cloudslang/content/excel/services/NewExcelDocumentService.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudslang.content.excel.services;
+
+import io.cloudslang.content.excel.entities.ExcelOperationException;
+import io.cloudslang.content.excel.entities.NewExcelDocumentInputs;
+import io.cloudslang.content.utils.OutputUtilities;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.FileOutputStream;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import static io.cloudslang.content.excel.services.ExcelServiceImpl.getFileFormat;
+import static io.cloudslang.content.excel.utils.Constants.BAD_CREATE_EXCEL_FILE_MSG;
+import static io.cloudslang.content.excel.utils.Constants.BAD_EXCEL_FILE_MSG;
+import static io.cloudslang.content.excel.utils.Constants.EXCEPTION_WORKSHEET_NAME_EMPTY;
+import static io.cloudslang.content.excel.utils.Constants.FORMAT_XLS;
+import static io.cloudslang.content.excel.utils.Constants.FORMAT_XLSX;
+import static io.cloudslang.content.utils.OutputUtilities.getFailureResultsMap;
+
+public class NewExcelDocumentService {
+
+    @NotNull
+    public static Map<String, String> newExcelDocument(@NotNull final NewExcelDocumentInputs newExcelDocumentInputs) {
+        final FileOutputStream output;
+        final String excelFileName = newExcelDocumentInputs.getExcelFileName();
+        final Workbook excelDoc;
+        try {
+            excelDoc = getNewExcelDocument(excelFileName);
+        } catch (Exception exception) {
+            return OutputUtilities.getFailureResultsMap(BAD_CREATE_EXCEL_FILE_MSG);
+        }
+
+        final String sheetnameDelimiter = newExcelDocumentInputs.getDelimiter();
+        String sheetNames = newExcelDocumentInputs.getWorksheetNames();
+
+        if (sheetNames.isEmpty()) {
+            sheetNames = "Sheet1" + sheetnameDelimiter + "Sheet2" + sheetnameDelimiter + "Sheet3";
+        }
+
+        final StringTokenizer tokenizer = new StringTokenizer(sheetNames, sheetnameDelimiter);
+        if (tokenizer.countTokens() == 0) {
+            return OutputUtilities.getFailureResultsMap(EXCEPTION_WORKSHEET_NAME_EMPTY);
+        }
+        try {
+            while (tokenizer.hasMoreTokens()) {
+                String sheetName = tokenizer.nextToken();
+                excelDoc.createSheet(sheetName);
+            }
+        } catch (Exception exception) {
+            return OutputUtilities.getFailureResultsMap(BAD_CREATE_EXCEL_FILE_MSG);
+        }
+
+        try {
+            output = new FileOutputStream(excelFileName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *output*. The resource is referenced in statements at the following line: 74. The resource closure statement is at line: 75. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Workbook.write. Either a) close *output* in a try-finally block or b) close the resource by declaring *output* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

Reply:
Done